Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.cpp

```diff
@@ -186,6 +186,12 @@ void WebExtensionContextProxy::setStorageAccessLevel(bool allowedInContentScript
     m_isSessionStorageAllowedInContentScripts = allowedInContentScripts;
 }
 
+void WebExtensionContextProxy::setContentScriptWorld(WebCore::DOMWrapperWorld& world)
+{
+    m_contentScriptWorld = world;
+    world.setShadowRootIsAlwaysOpen();
```

Comment:
I don't think we want to set this option if we're adding a new API to access closed shadow roots.
This API also changes the behavior of event.composedPath, etc...

Reply:
Do other browser really expose nodes in `event.composedPath`? I don't think we want to diverge from other browsers here.

Reply:
Chrome and Firefox have said that are planning to do it in when it came up in the WECG.

Reply:
Okay, in that case, we should add a new DOM wrapper option for extensions and expose nodes in closed shadow trees when the option is set in addition to exposing new API to extensions.

But we definitely don't want to expose `collectMatchingElementsInFlatTree` or `matchingElementInFlatTree`.

Reply:
`shadowRootIsAlwaysOpen` also changes the behavior of `Element.prototype.shadowRoot` and I don't think we want to do that either.

Reply:
Alright. I'll revise then. Thanks!